src/page/page_cache.rs

```diff
@@ -99,13 +100,12 @@ impl<PM, P, R> PageCache<PM, LockFreeLog, P, R>
 
     /// Create a new page, trying to reuse old freed pages if possible
     /// to maximize underlying `Radix` pointer density.
-    pub fn allocate(&self) -> (PageID, *const stack::Node<CacheEntry<P>>) {
+    pub fn allocate(&self) -> (PageID, CasKey<P>) {
```

Comment:
all public-facing api's have no idea about coco scopes, or the details of the pointers that will later be used for CAS operations. the details are hidden away in the `CasKey` that they do have access to.